platforms/web/lib/useFormattingActions.ts

```diff
@@ -17,20 +17,15 @@ limitations under the License.
 import { RefObject, MouseEvent as ReactMouseEvent, useMemo } from 'react';
 
 import { BlockType } from './types';
-import { sendInputEvent } from './useListeners';
+import { sendFormatBlockEvent } from './useListeners';
 
 export function useFormattingActions(editorRef: RefObject<HTMLElement | null>) {
     const formattingActions = useMemo(() => {
-        const sendEvent = (e: ReactMouseEvent<HTMLElement, MouseEvent> | KeyboardEvent,
-            inputType: InputEvent['inputType']) => editorRef.current && sendInputEvent(e, editorRef.current, inputType);
-
         // The formatting action like inline code doesn't have an input type
-        // For this case, we send a custom event and we do not use the browser input event handling
-        const sendFormatBlockEvent = (e: ReactMouseEvent<HTMLElement, MouseEvent>, blockType: BlockType) => {
-            e.preventDefault();
-            e.stopPropagation();
-            editorRef.current?.dispatchEvent(new CustomEvent('formatBlock', { detail: { blockType } }));
-        };
+        // Safari does not keep the inputType in an input event when the input event is fired manually
+        // So we send a custom event and we do not use the browser input event handling
+        const sendEvent = (e: ReactMouseEvent<HTMLElement, MouseEvent> | KeyboardEvent,
+            blockType: BlockType) => editorRef.current && sendFormatBlockEvent(e, editorRef.current, blockType);
```

Comment:
sendFormatBlockEvent should be renamed now, right?

Reply:
It sends the `formatBlock` event, for me the name is correct.

Rename the `formatBlock` event is more relevant in my opinion. I don't have a more suitable name in my mind ^^

Reply:
Doesn't it send all types of event now, not just formatBlock?

If formatBlock now means all events, that should be renamed to `ourInput` or similar I guess.